Update documentation about downloading CodeQL databases using the VS Code extension (#10270)
github/codeql: docs/codeql/reusables/download-github-database.rst

GitHub also stores CodeQL databases for over 200,000 repos on GitHub.com, which you can download using the API. The list of repos is constantly growing and evolving to make sure that it includes the most interesting codebases for security research.

> 

Reviewer: I think we're going to want to link out to the new download CodeQL database endpoint here, once the OpenAPI changes are published. Presumably that content will explain what the acceptable values for `<language>` are?

It may be worth giving a little more information about the endpoint here, since the GitHub CLI isn't the only way to download a database from GitHub.com. Then just use the GH CLI example below as an example of a way to use the endpoint.

To download a database from GitHub.com using the `GitHub CLI <https://cli.github.com/manual/gh_api>`__, use the following command:: 

> 

Reviewer: Is there any way to find out whether or not a CodeQL database is available for a project, either in the GitHub UI or via a REST API call?

GitHub stores CodeQL databases for over 200,000 repos on GitHub.com, which you can download using the REST API. The list of repos is constantly growing and evolving to make sure that it includes the most interesting codebases for security research.

> You can check if a repo has any CodeQL databases available to download using the `/repos/<owner>/<repo>/code-scanning/codeql/databases` endpoint.

Reviewer: ```suggestion
You can check if a repository has any CodeQL databases available for download using the ``/repos/<owner>/<repo>/code-scanning/codeql/databases`` endpoint.
```

For example, to check for CodeQL databases using the `GitHub CLI <https://cli.github.com/manual/gh_api>`__ you would run::

   gh api /repos/<owner>/<repo>/code-scanning/codeql/databases/

> This command will return information about the CodeQL databases that are available for a repository, including the language the database represents, and when the database was last updated. If no CodeQL databases are available, the response will be empty.

Reviewer: ```suggestion
This command returns information about any CodeQL databases that are available for a repository, including the language the database represents, and when the database was last updated. If no CodeQL databases are available, the response is empty.
```
